File: src/main/java/io/github/mavenindexdoctor/domain/MavenDiagnosticReport.java
package io.github.mavenindexdoctor.domain;

import java.nio.file.Path;
import java.time.Instant;
import java.util.List;

public record MavenDiagnosticReport(
        Instant scannedAt,
        Path repositoryPath,
        Path indexPath,
        Path settingsPath,
        int localArtifactCount,
        int missingFromIndexCount,
        List<DiagnosticIssue> issues
) {
}

File: src/main/kotlin/io/github/mavenindexdoctor/domain/MavenDiagnosticReport.kt
package io.github.mavenindexdoctor.domain

import java.time.Instant
import java.nio.file.Path

data class MavenDiagnosticReport(
    val scannedAt: Instant,
    val repositoryPath: Path,
    val indexPath: Path,
    val issues: List<DiagnosticIssue>,
)
